cypress/support/step_definitions/noTrials.js

```diff
@@ -0,0 +1,165 @@
+/// <reference types="Cypress" />
+import { Given, When, Then, And } from 'cypress-cucumber-preprocessor/steps';
+
+// Setup for each test
+beforeEach(() => {
+  // This ensures we wait for any async operations to complete
+  cy.window().then(win => {
+    if (!win.INT_TEST_APP_PARAMS) {
+      win.INT_TEST_APP_PARAMS = {};
+    }
+  });
+});
+
+// Steps for data entry in filters
+Given('user has typed {int} in the Age text input', (age) => {
+  cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+  cy.get('.ctla-sidebar').find('input[type="number"]').clear().type(age);
+});
+
+Given('user has typed {string} in the Location by ZIP Code text input', (zipCode) => {
+  cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+  cy.get('.ctla-sidebar').find('input[type="text"]').clear().type(zipCode);
+});
+
+Given('user has selected {int} miles for the Radius dropdown', (radius) => {
+  cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+  cy.get('.ctla-sidebar').find('select.usa-select').select(radius.toString());
+});
+
+// Mock API responses for consistent testing
+Given('the API returns empty results for filtered search', () => {
+  cy.intercept('POST', '*/trials*', {
+    statusCode: 200,
+    body: {
+      total: 0,
+      data: []
+    }
+  }).as('emptyResults');
+});
+
+// URL validation
+Then('the page URL stays the same with the additional query parameter', () => {
+  cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+  cy.url().should('include', 'a='); // Age filter parameter
+  cy.url().should('include', 'z='); // ZIP code parameter
+  cy.url().should('include', 'zr='); // Radius parameter - Fixed from 'r=' to 'zr='
+});
+
+// Content validation
+Then('intro text displays', () => {
+  cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+  cy.get('.ctla-results__intro').should('be.visible');
+});
+
+Then('Trials 0 of 0 displays below the intro text', () => {
```

Comment:
looks like this step is duplicated in 5 different files - why dont you put it in the common folder index.js ? it will pick up the step definition thats in the common. If you need to override the common folder step, you can add it to respective feature folder, but only if the code is different.

Reply:
this is true for all the steps that are duplicated across the multiple files - you should only need to add it once into common/index.js and all features will work